ibm/service/satellite/data_source_ibm_satellite_host_script.go

```diff
@@ -126,51 +131,72 @@ func dataSourceIBMSatelliteAttachHostScriptRead(d *schema.ResourceData, meta int
 		}
 	}
 	scriptDir, _ = filepath.Abs(scriptDir)
-	scriptPath := filepath.Join(scriptDir, "addHost.sh")
+	var scriptPath string
 
 	//Generate script
 	createRegOptions := &kubernetesserviceapiv1.AttachSatelliteHostOptions{}
 	createRegOptions.Controller = locData.ID
 	createRegOptions.Labels = labels
 
+	//check to see if host attach is CoreOS or RHEL
+	var host_os string
+	var coreos_enabled bool
+	if _, ok := d.GetOk("coreos_host"); ok {
+		coreos_enabled = d.Get("coreos_host").(bool)
+		if coreos_enabled {
+			host_os = "RHCOS"
+			createRegOptions.OperatingSystem = &host_os
+			scriptPath = filepath.Join(scriptDir, "addHost.ign")
+		}
+	} else {
```

Comment:
I suspect if it enters this else case at all.. as GetOk on boolean will always have some value as default value for boolean is false..

Can we use `d.GetOkExists("coreos_host")` instead of `d.GetOk("coreos_host")`

Reply:
I tested it both with `coreos_host` = false and no `coreos_host` flag - and both times it provided the correct RHEL script.

If I'm understanding the test: `func testAccCheckIBMSatelliteAttachHostScriptDataSourceConfig` - that should show it, as `coreos_host` is not presented, and it returns the correct script.

The test itself doesn't show the contests of the attach script - but locally I can see the correct scripts provided. I'll upload the results here.